proposals/semi-auto-properties.md

```diff
@@ -1,14 +1,307 @@
+
+Here's an example of where this is useful. We believe the `field` keyword will find a lot of its use with view models because of the elegant solution it brings for the `INotifyPropertyChanged` pattern. View model property setters are likely to be databound to UI and likely to cause change tracking or trigger other behaviors. The following code needs to initialize the default value of `IsActive` without setting `HasPendingChanges` to `true`:
+
+```cs
+class SomeViewModel
+{
+    public bool HasPendingChanges { get; private set; }
+
+    public bool IsActive { get; set => Set(ref field, value); } = true;
+
+    private bool Set<T>(ref T location, T value)
+    {
+        if (RuntimeHelpers.Equals(location, value))
+            return false;
+
+        location = value;
+        HasPendingChanges = true;
+        return true;
+    }
+}
+```
+
+This difference in behavior between a property initializer and assigning from the constructor can also be seen with virtual auto properties in previous versions of the language:
+
+```cs
+using System;
+
+// Nothing is printed; the property initializer is not
+// equivalent to `this.IsActive = true`.
+_ = new Derived();
+
+class Base
+{
+    public virtual bool IsActive { get; set; } = true;
+}
+
+class Derived : Base
+{
+    public override bool IsActive
+    {
+        get => base.IsActive;
+        set
+        {
+            base.IsActive = value;
+            Console.WriteLine("This will not be reached");
+        }
+    }
+}
+```
+
+### Constructor assignment
+
+As with auto properties, assignment in the constructor calls the setter if it exists, and if there is no setter it falls back to directly assigning to the backing field.
+
+```cs
+class C
+{
+    public C()
+    {
+        P1 = 1; // Assigns P1's backing field directly
+        P2 = 2; // Assigns P2's backing field directly
+        P3 = 3; // Calls P3's setter
+        P4 = 4; // Calls P4's setter
+    }
+
+    public int P1 => field;
+    public int P2 { get => field; }
+    public int P4 { get => field; set => field = value; }
+    public int P3 { get => field; set; }
+}
+```
+
+### Definite assignment in structs
+
+Even though they can't be referenced in the constructor, backing fields denoted by the `field` keyword are subject to default-initialization and disabled-by-default warnings under the same conditions as any other struct fields ([LDM decision 1](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-03-02.md#property-assignment-in-structs), [LDM decision 2](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-05-02.md#definite-assignment-of-manually-implemented-setters)). For example:
+
+```cs
+public struct S
+{
+    public S()
+    {
+        _ = P1; // Disabled-by-default warning
+        P2 = 5; // Disabled-by-default warning
+    }
```

Comment:
When I read "disabled-by-default warning", I initially interpreted that as a warning for a feature named "disabled-by-default". Consider rewording, and perhaps include the warning. Also, what is the warning for the second statement `P2 = 5;`?

Perhaps:
```
_ = P1; // warning CS9018: P1 is read before assignment (warning is disabled by default)
P2 = 5; // ... what is the warning?
```

Reply:
The sample muddied the waters a bit. But imagine the body only contains one or the other of these statements. In that case we need to implicitly assign the fields of 'this' and will report diagnostics (which 99.99% of users will leave disabled by default)

Reply:
I separated into two samples and documented the exact diagnostics.